expo/expo
# [splash-screen][dev-launcher] fix cold boot deep link error handler VC

# Why



This is a fix for an error Jon reported about deep links not opening on the dev launcher - note that this happens in a cold boot, meaning the app isn't open or in the background!

The issue appears to be two fold. First, the view controller that renders errors captured when handling invalid deep links was not visible. Second, the splash screen package listens for the same `rootViewController` assignment used by the error reporting VC, and renders on top of this view, blocking the error view from being visible

# How



The first issue was solved by adding `makeKeyAndVisible()` to the window that is rendering the error view - the symptom of this not being present is an entirely black screen (as if there were no window at all ) which makes sense.

The second issue was solved by manually hiding the splash screen - in order to do this, I've added a notification that the splash screen package listens for - I'm not sure if this is the best solution but it's the simplest one I could think of. 

# Test Plan



- Pull in the repository Jon posted in the Linear issue
- run `expo prebuild` 
- symlink the splash screen and `dev-launcher` packages
- run `pod install` and ensure the packages have the changes from this branch 
- build the app once locally (if running on simulator)
- ensure the app is closed
- update the launcher settings for the build in Xcode: 

<img width="552" alt="Screen Shot 2022-04-11 at 12 35 09 PM" src="https://user-images.githubusercontent.com/40680668/162816258-972d6e62-5641-4532-bd20-292579b89f5a.png">

- deep link into the app using the scheme (this has an invalid link encoded to prompt the error handler):
`exp+test-4-6-dev-client-classic://expo-development-client?url=https%3A%2F%2Fu.expo.dev%2F12d3133b-6509-4932-be8c-2dda6f48d51c`
- ensure that the error screen shows up

- deep link into the app when its in the background
- ensure that the error screen shows up

- deep link into app when its in the foreground
- ensure the error screen shows up (this one is unlikely to happen, but can be achieved w/ simulator)

### Other Stuff

@esamelson could this possibly have compatibility issues with the new `ExpoModuleProviders` patterns? I have some issues trying to get the `onDeepLink` method to be called, I suspect its because I can't quite get the right local setup configured 

This also should fix the issue where the splash screen shows over top of the error handler for invalid links while the app is in background (or foreground, for that matter), which would happen to me somewhat frequently while developing dev launcher (I never quite understood why until now!)

# Checklist
